File: sample/venomtyme.ex
defmodule Cat do
  @moduledoc """
  Once upon a time...
  """

  defstruct [:location, :birth_date, :death_date, :catlikes]

  def new(props) do
    %__MODULE__{
      location: props[:location],
      birth_date: props[:birth_date],
      death_date: props[:death_date],
      catlikes: props[:catlikes]
    }
  end
  
  def age(cat) do
    calc_age(cat)
  end

  defp calc_age(cat) do
    cat.death_date - cat.birth_date
  end
end

# ...there was a cat named Katlyn

catppuccin = Cat.new(
  location: "Cataliputra",
  birthDate: 2016,
  currentDate: 2022,
  catlikes: ["Tasty Food", "Long Naps"]
)

File: sample/venomtyme.dart
/*
 * Once upon a time...
 */

class Cat {
  String location;
  int birthDate, currentDate;
  List<String> catlikes;

  Cat({this.location, this.birthDate, this.currentDate, this.catlikes});

  int get age => this.calcAge();

  int calcAge() => this.currentDate - this.birthDate;
}

void main() {
  // ...there was a cat named Katlyn
  final Catppuccin = Cat(
      location: 'Cataliputra',
      birthDate: 2016,
      currentDate: 2022,
      catlikes: ['Tasty Food', 'Long Naps']);
}